pkg/initializers/dataset/cache.py

```diff
@@ -0,0 +1,295 @@
+import logging
+import time
+from typing import Optional
+
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+
+import pkg.initializers.types.types as types
+import pkg.initializers.utils.utils as utils
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%SZ",
+    level=logging.INFO,
+)
+
+
+class CacheInitializer(utils.DatasetProvider):
+
+    def load_config(self):
+        config_dict = utils.get_config_from_env(types.CacheDatasetInitializer)
+        self.config = types.CacheDatasetInitializer(**config_dict)
+
+    def download_dataset(self):
+        logging.info(
+            f"Cache initializer called with storage URI: {self.config.storage_uri}"
+        )
+
+        train_job_name = self.config.train_job_name
+        cache_image = self.config.cache_image
+        cluster_size = int(self.config.cluster_size)
+        iam_role = self.config.iam_role
+        head_cpu = self.config.head_cpu
+        head_mem = self.config.head_mem
+        worker_cpu = self.config.worker_cpu
+        worker_mem = self.config.worker_mem
+
+        # Create Kubernetes resources using client SDK
+        create_cache_resources(
+            train_job_name=train_job_name,
+            iam_role=iam_role,
+            cluster_size=cluster_size,
+            cache_image=cache_image,
+            head_cpu=head_cpu,
+            head_mem=head_mem,
+            worker_cpu=worker_cpu,
+            worker_mem=worker_mem,
+            namespace=self.config.namespace,
+            metadata_loc=self.config.metadata_loc,
+            table_name=self.config.table_name,
+            schema_name=self.config.schema_name,
+        )
```

Comment:
Rather than splitting this between two functions, I would keep everything in `create_cache_cluster()` to reduce amount of code.